### ISSUE TITLE
Selected fields are not available in Info for Fragments

### ISSUE THREAD

**Issue by cbergmiller** [strawberry-graphql/strawberry#1197]
The following test shows that the selected fields for the Fragment `frag` are not available in `Info.selected_fields`:
```python
def test_info_field_fragment():
    @strawberry.type
    class Result:
        field: str

    @strawberry.type
    class Query:
        @strawberry.field
        def hello(self, info: Info[str, str]) -> Result:
            print(info.selected_fields)
            return Result(field=json.dumps(dataclasses.asdict(*info.selected_fields)))

    schema = strawberry.Schema(query=Query)
    query = """
    fragment frag on Result {
        field
    }
    query {
        hello {
            ...frag
        }
    }
    """
    result = schema.execute_sync(query)
    assert not result.errors
    info = result.data["hello"]
    field = json.loads(info["field"])
    assert field == {
        "name": "hello",
        "directives": {},
        "alias": None,
        "arguments": {},
        "selections": [
            {"name": "frag", "directives": {}},
        ],
    }
```

`selected_fields` is `[SelectedField(name='hello', directives={}, alias=None, arguments={}, selections=[FragmentSpread(name='frag', directives={})])]`
Is this the intended behaviour? I am in the process of rewriting [graphene-django-optimizer](https://github.com/tfoxy/graphene-django-optimizer) to work with strawberry-django. For this use case it would be very helpfull to have the selected fields of the Fragment available in `Info.selected_fields`.

**Comment by jkimbo** [comment]
@cbergmiller I've created a PR to fix this: #1213

**Comment by cbergmiller** [comment]
@jkimbo i ran the test and inspected `Info.selected_fields` - looks great, thanks!
